src/Data/Parity/Properties.agda

```diff
@@ -420,58 +417,137 @@ toSign-injective {p} {q} eq = begin
   ; injective = toSign-injective
   }
 
-toSign-isMonoidIsomorphism : IsMonoidIsomorphism +-0-rawMonoid Sign.*-1-rawMonoid toSign
+toSign-isMonoidIsomorphism : IsMonoidIsomorphism ℙ.+-0-rawMonoid 𝕊.*-1-rawMonoid toSign
 toSign-isMonoidIsomorphism = record
   { isMonoidMonomorphism = toSign-isMonoidMonomorphism
   ; surjective = toSign-surjective
   }
 
-toSign-isGroupHomomorphism : IsGroupHomomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupHomomorphism : IsGroupHomomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupHomomorphism = record
   { isMonoidHomomorphism = toSign-isMonoidHomomorphism
   ; ⁻¹-homo = ⁻¹-homo-opposite
   }
 
-toSign-isGroupMonomorphism : IsGroupMonomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupMonomorphism : IsGroupMonomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupMonomorphism = record
   { isGroupHomomorphism = toSign-isGroupHomomorphism
   ; injective = toSign-injective
   }
 
-toSign-isGroupIsomorphism : IsGroupIsomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupIsomorphism : IsGroupIsomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupIsomorphism = record
   { isGroupMonomorphism = toSign-isGroupMonomorphism
   ; surjective = toSign-surjective
   }
 
+
+------------------------------------------------------------------------
+-- relating Nat and Parity
+
+-- successor and (_⁻¹)
+
+suc-homo-⁻¹ : ∀ n → (parity (suc n)) ⁻¹ ≡ parity n
+suc-homo-⁻¹ zero    = refl
+suc-homo-⁻¹ (suc n) = ⁻¹-inverts (suc-homo-⁻¹ n)
+
+-- parity is a _+_ homomorphism
+
++-homo-+      : ∀ m n → parity (m ℕ.+ n) ≡ parity m ℙ.+ parity n
++-homo-+ zero    n = refl
++-homo-+ (suc m) n = begin
+  parity (suc m ℕ.+ n)        ≡⟨ suc-+-homo-⁻¹ m n ⟩
+  (parity m) ⁻¹ ℙ.+ parity n   ≡⟨ cong (ℙ._+ parity n) (suc-homo-⁻¹ (suc m)) ⟩
+  parity (suc m) ℙ.+ parity n  ∎
+  where
+    open ≡-Reasoning
+    suc-+-homo-⁻¹ : ∀ m n → parity (suc m ℕ.+ n) ≡ (parity m) ⁻¹ ℙ.+ parity n
+    suc-+-homo-⁻¹ zero    n = sym (suc-homo-⁻¹ (suc n))
+    suc-+-homo-⁻¹ (suc m) n = begin
+      parity (suc (suc m) ℕ.+ n)        ≡⟨⟩
+      parity (m ℕ.+ n)                  ≡⟨ +-homo-+ m n ⟩
+      parity m ℙ.+ parity n              ≡⟨ cong (ℙ._+ parity n) (sym (suc-homo-⁻¹ m)) ⟩
+      (parity (suc m)) ⁻¹ ℙ.+ (parity n) ∎
+
+-- parity is a _*_ homomorphism
+
+*-homo-*      : ∀ m n → parity (m ℕ.* n) ≡ parity m ℙ.* parity n
+*-homo-* zero    n = refl
+*-homo-* (suc m) n = begin
+  parity (suc m ℕ.* n)       ≡⟨⟩
+  parity (n ℕ.+ m ℕ.* n)    ≡⟨ +-homo-+ n (m ℕ.* n) ⟩
+  q ℙ.+ parity (m ℕ.* n)     ≡⟨ cong (q ℙ.+_) (*-homo-* m n) ⟩
+  q ℙ.+ (p ℙ.* q)             ≡⟨ lemma p q ⟩
+  ((p ⁻¹) ℙ.* q)              ≡⟨⟩
+  ((parity m) ⁻¹ ℙ.* q)       ≡⟨ cong (ℙ._* q) (suc-homo-⁻¹ (suc m)) ⟩
+  parity (suc m) ℙ.* q        ≡⟨⟩
+  parity (suc m) ℙ.* parity n ∎
+  where
+    open ≡-Reasoning
+    p = parity m
+    q = parity n
+    -- this lemma simplifies things a lot
```

Comment:
Same for this lemma, pull it out. It feels like it ought to be useful in other proofs as well.

Reply:
Perhaps you're right. It didn't 'smell' right to me as a property of independent interest. I think that it follows 'easily' (at least: I believe it does, but *laboriously*) from `p+p≡0ℙ`, `p⁻¹+p≡1ℙ`, `*-identityˡ`, and `*-distribˡ-+` , but the calculational proof here was definitely quicker/slicker...

Reply:
I've been there before: a property that doesn't seem to naturally arise from the external API, but is somehow key in a proof and general.

My experience has been that such key lemmas show up again later, so I pull them out. [I still have multiple passes to do on agda-categories to refactor multiple proofs to use the patterns that were discovered after the proofs were laboriously hand-written.]

It's a very subjective call.